doc/memos/rdm-draft-petry-riot-design-goals.md

```diff
@@ -0,0 +1,218 @@
+to continue operating if a user is on holiday. It should be able to send fairly
+continuous data over a widely supported wireless LAN protocol, in order to
+support commercial home gateways.
+
+## Communication daughterboards for low-power protocols
+
+This requires RIOT to support popular emerging low-power protocols, and to
+provide robust and stable support for commonly used microcontrollers and
+transcievers.
+
+## Shipment tracking
+
+This requires a node to last for a number of weeks without recharge, monitor a
+range of environmental conditions, and send data to a local gateway which will
+be connected to the vehicle's power source and handle GPS localization.
+
+## Wearables
+
+This requires a node to be extremely light and low power, and to send data to a
+public gateway over a very low bandwidth connection, for example LoRaWAN, or to
+the user's smartphone, for example over Bluetooth. The device would require
+charging perhaps every week.
+
+## Diagnostic collection and/or control of traditionally non-powered devices
+
+This requires a node to be able to be retrofitted to a non-powered device such
+as a personal transportation device or item of public infrastructure, to be able
+to harvest energy from that device or last for a number of weeks or months
+without charging, to perform sensing of environmental or mechanical parameters,
+and to send data to a user's phone or a public wide-area network.
+
+## Building heating, ventilation, and air conditioning (HVAC)
+
+This requires a node to be able to be retrofitted to an already existing
+building, which may involve wireless communication but will likely involve a
+provided power source to the node. It will involve sensing of environmental
+conditions such as temperature, air quality and composition, and be able to
+feed back data at a rate that allows control of those conditions.
+
+
+# 3. List of design philosophies
+
+Below are a list of our design philosophies, including, where appropriate,
+an indication of relative priorities.
+
+## Energy efficiency
+
+RIOT nodes should be able to survive for years without running out of power.
+This means that they need to be able to be extremely conservative with power,
+and harvest any available energy from the environment.
+
+## Real-time capabilities
+
+For non-control and low-frequency sensing applications, precise knowledge of the
+data capture time is not neccessary. Nevertheless, RIOT users may wish to use
+the operating system in control or high frequency data capture applications, and
+in such a case, harder real-time guarantees and more accurate timing should be
+achievable at the expense of power consumption and configuration difficulty.
+
+## Small memory footprint
+
+RIOT aims to support constrained devices in class 1 [1]. This means that
+RIOT applications, including their libraries, consume ~ 10 KiB of RAM and ~100
+KiB of ROM, or less. All modules provided by RIOT can be used practically in
```

Comment:
This is not true and also not desirable. We want core components to stay under certain (which ?) memory limits and still support larger modules.

Reply:
I've rephrased this paragraph to reflect this.